pandas/io/pytables.py

```diff
@@ -404,7 +404,7 @@ def read_hdf(path_or_buf, key=None, mode="r", **kwargs):
                     )
             key = candidate_only_group._v_pathname
         return store.select(key, auto_close=auto_close, **kwargs)
-    except (ValueError, TypeError, KeyError):
+    except Exception:
```

Comment:
there is an ongoing effort to avoid catching Exception.  Can we specify what additional ways this can fail?

Reply:
`/pandas/core/internals/managers.py`, in `_rebuild_blknos_and_blklocs` can raise an `AssertionError`. That was my initial problem.

I do understand the need of avoiding catching `Exception` but in this specific case, since we re-raise whatever exception occurs, it did not feel that bad.

Reply:
Can the `close` operation just be defined in a `finally` block here instead?

Reply:
> Can the `close` operation just be defined in a `finally` block here instead?

Indeed, and thus completly remove the `except` block that would become useless. I'm perfectly fine with it, this seems like the most explicit and natural way to (not-) handle errors here.

Reply:
That would probably be the preferred fix here.